superset/connectors/sqla/models.py

```diff
@@ -1247,35 +1254,45 @@ def fetch_metadata(self, commit: bool = True) -> None:
         any_date_col = None
         db_engine_spec = self.database.db_engine_spec
         db_dialect = self.database.get_dialect()
-        dbcols = (
-            db.session.query(TableColumn)
-            .filter(TableColumn.table == self)
-            .filter(or_(TableColumn.column_name == col.name for col in table_.columns))
```

Comment:
I'm not sure what the intent here was, but I don't see any scenario where we want to bring in columns from another table with the same column names.